test/dynamics/solvers/test_solver_classes.py

```diff
@@ -67,80 +141,267 @@ def test_validation_override_lindblad(self):
 
         self.assertAllClose(solver.model.hamiltonian_operators, operators)
 
+    def test_incompatible_input_lengths(self):
```

Comment:
Consider testing single length args? make sure they convert to the right size

Reply:
This is actually tested in the `TestSolverListSimulation` class further down in the file. I put a bunch of tests there for making sure various configurations of lists/singletons work.

Reply:
Perfect!